src/main/java/by/exadel/internship/repository/UserRepository.java

```diff
@@ -25,9 +28,14 @@
     List<User> findAllByDeletedFalse();
 
     Optional<User> findByIdAndDeletedTrue(UUID userId);
-
     Optional<User> findByIdAndDeletedFalse(UUID userId);
 
+    @EntityGraph(attributePaths = {"skills", "userTimeSlots"})
+        @Query("SELECT DISTINCT u FROM User  u LEFT JOIN u.userTimeSlots  ts " +
+                "WITH ts.startDate   >= CAST (CURRENT_TIMESTAMP as org.hibernate.type.LocalDateTimeType) WHERE  ((u.id = :userId ) AND u.deleted = false) ")
```

Comment:
why do you use [`WITH`](https://www.geeksforgeeks.org/sql-with-clause/)? Move it to `where` section. And remove excessive brackets

Reply:
If I move it In the section where the user won't be found, because I need to cut timeslots, but to find user(in case if user has only overdue timeslots) , othewise condition "where" cut  user as well/
 excessive braces were deleted